File: app/src/main/java/com/test/jarustest/model/ParentAssignment.java
package com.test.jarustest.model;

import com.bignerdranch.expandablerecyclerview.Model.ParentListItem;
import com.test.jarustest.model.ChildAssignment;

import java.util.List;

public class ParentAssignment implements ParentListItem {
    private final int mId;
    private final String mTitle;
    private boolean isChecked;
    private List<ChildAssignment> mChildItemList;

    public ParentAssignment(int mId, String mTitle) {
        this.mId = mId;
        this.mTitle = mTitle;
    }

    public boolean isChecked() {
        return isChecked;
    }

    public void setChecked(boolean checked) {
        isChecked = checked;
    }

    public int getMId() {
        return mId;
    }

    public String getMTitle() {
        return mTitle;
    }

    @Override
    public List<ChildAssignment> getChildItemList() {
        return mChildItemList;
    }

    @Override
    public boolean isInitiallyExpanded() {
        return false;
    }

    public void setChildItemList(List<ChildAssignment> list) {
        mChildItemList = list;
    }
}

File: app/src/main/java/com/test/jarustest/model/ParentAssignment.kt
package com.test.jarustest.model

import com.bignerdranch.expandablerecyclerview.Model.ParentListItem

class ParentAssignment(val mId: Int, val mTitle: String) : ParentListItem {
    var isChecked = false
    private var mChildItemList: List<ChildAssignment?>? = null

    override fun getChildItemList(): List<ChildAssignment?> {
        return mChildItemList!!
    }

    override fun isInitiallyExpanded(): Boolean {
        return false
    }

    fun setChildItemList(list: List<ChildAssignment?>?) {
        mChildItemList = list
    }

}